### PR TITLE
Detections Inheritance

### PR DESCRIPTION
**Pull request by grantjoy**
### Background

Adds support for derived detections.

### Testing

* Manual testing. Doesn't affect detections at all unless they have BaseDetection set, which would just be the new inheritance detections only.

### CODE REVIEW

File: panther_analysis_tool/schemas.py [hunk 1]
```diff
@@ -154,44 +154,95 @@ def validate(
...
+    Optional("AlertTitle"): str,
+    Optional("AlertContext"): object,
+    Optional("GroupBy"): object,
+    Optional("BaseDetection"): And(str, NAME_ID_VALIDATION_REGEX),
+}
+
+RULE_SCHEMA = Schema(dict_schema, ignore_extra_keys=False)
+
+
+def validate_rule(data):
```
Comment (stedrow) [inline panther_analysis_tool/schemas.py:241]: is there ever invoked? I'm not seeing where we're actually running this

File: panther_analysis_tool/schemas.py [hunk 2]
```diff
@@ -154,44 +154,95 @@ def validate(
...
+            "Enabled",
+            "Severity",
+            "Filename",
+            "Detection",
+            "LogTypes",
+            "ScheduledQueries",
+        }
+        for field in base_required_fields:
+            if field not in d:
+                raise SchemaError(f"'{field}' is required when 'BaseDetection' is not present.")
```
Comment (darwayne) [inline panther_analysis_tool/schemas.py:175]: I wouldn't even mention BaseDetection in this context.

Comment (grantjoy) [reply panther_analysis_tool/schemas.py:175]: thanks. removing all this.

File: panther_analysis_tool/schemas.py [hunk 3]
```diff
@@ -193,6 +193,33 @@ def validate(
...
+        Optional("SummaryAttributes"): [str],
+        Optional("Threshold"): int,
+        Optional("Tags"): [str],
+        Optional("Reports"): {str: list},
+        Optional("DynamicSeverities"): object,
+        Optional("AlertTitle"): str,
+        Optional("AlertContext"): object,
+        Optional("GroupBy"): object,
+    },
+    ignore_extra_keys=False,
```
Comment (darwayne) [inline panther_analysis_tool/schemas.py:220]: I don't see this being used anywhere I think you'll need to update [this section](https://github.com/panther-labs/panther_analysis_tool/blob/6b45e21d132e3a0d59029d4509d6e920984c5a43/panther_analysis_tool/main.py#L2030) to include the derived schema and set to true conditionally as we seem to do for other types

File: panther_analysis_tool/schemas.py [hunk 4]
```diff
@@ -193,6 +193,33 @@ def validate(
     ignore_extra_keys=False,
 )  # Prevent user typos on optional fields
 
+DERIVED_SCHEMA = Schema(
+    {
+        "AnalysisType": Or("rule", "scheduled_rule"),
```
Comment (darwayne) [inline panther_analysis_tool/schemas.py:198]: this should just be rule for now

**Review by darwayne — COMMENTED** [review]
Nice stuff, think we're almost there

File: panther_analysis_tool/schemas.py (hunk 3 above)
Comment (grantjoy) [reply panther_analysis_tool/schemas.py:220]: Added, thanks.

File: panther_analysis_tool/schemas.py (hunk 4 above)
Comment (grantjoy) [reply panther_analysis_tool/schemas.py]: changed, thanks.

**Review by darwayne — APPROVED** [review]

**Pull request merged**